File: app/src/main/java/com/scop/org/minesweeper/skins/NostalgiaSkin.java
package com.scop.org.minesweeper.skins;

import android.content.Context;
import android.graphics.Bitmap;
import android.graphics.Canvas;

import com.scop.org.minesweeper.R;

public class NostalgiaSkin extends MainSkin {
	private static final int ALTERNATIVES_MAX = 1;
	private Bitmap empty;
	private Bitmap[] covers;

	@Override
	public void load(Context context) {
		bgColor = 0xFF3C3C3C;
		empty = getBitmap(context, R.drawable.tile_skin_win_empty);

		symbols[0] = getBitmap(context, R.drawable.tile_skin_win_bomb);
		symbols[1] = getBitmap(context, R.drawable.tile_skin_win_bomb_end);
		symbols[2] = getBitmap(context, R.drawable.tile_skin_win_flag);
		symbols[3] = getBitmap(context, R.drawable.tile_skin_win_flag_fail);

		for (int i=1; i<9; i++) {
			numbers.put(i+"_"+i,getBitmap(context, "tile_skin_win_"+i+"_"+i));
		}

		tileSize = symbols[0].getHeight();

		Bitmap mainCover = getBitmap(context, R.drawable.tiles_win);
		int coverH = mainCover.getHeight();
		int coverW = mainCover.getWidth()/ALTERNATIVES_MAX;
		covers = new Bitmap[ALTERNATIVES_MAX];
		for (int i = 0; i<ALTERNATIVES_MAX; i++) {
			covers[i] = Bitmap.createBitmap(mainCover, i*coverW, 0, coverW, coverH);
		}
	}

	@Override
	public void drawCovered(Canvas canvas, float x, float y) {
		int i = alternative % covers.length;
		canvas.drawBitmap(covers[i], x, y, defaultPaint);
	}

	@Override
	public void drawEmpty(Canvas canvas, float x, float y) {
		canvas.drawBitmap(empty, x, y, defaultPaint);
	}
}

File: app/src/main/java/org/oar/minesweeper/skins/NostalgiaSkin.kt
package org.oar.minesweeper.skins

import android.content.Context
import android.graphics.Bitmap
import android.graphics.Canvas
import org.oar.minesweeper.R

class NostalgiaSkin : MainSkin() {

    override var bgColor = -0xc3c3c4
    override lateinit var symbols: Array<Bitmap>

    private lateinit var empty: Bitmap
    private lateinit var covers: Array<Bitmap>

    override fun load(context: Context) {
        empty = getBitmap(context, R.drawable.tile_skin_win_empty)
        symbols = arrayOf(
            getBitmap(context, R.drawable.tile_skin_win_bomb),
            getBitmap(context, R.drawable.tile_skin_win_bomb_end),
            getBitmap(context, R.drawable.tile_skin_win_flag),
            getBitmap(context, R.drawable.tile_skin_win_flag_fail))

        for (i in 1..8) {
            numbers[i.toString() + "_" + i] = getBitmap(context, "tile_skin_win_" + i + "_" + i)
        }
        tileSize = symbols[0].height - 1
        val mainCover = getBitmap(context, R.drawable.tiles_win)
        val coverH = mainCover.height
        val coverW = mainCover.width

        covers = arrayOf(
            Bitmap.createBitmap(mainCover, 0, 0, coverW, coverH))
    }

    override fun drawCovered(canvas: Canvas, x: Float, y: Float) {
        val i = alternative % covers.size
        canvas.drawBitmap(covers[i], x, y, defaultPaint)
    }

    override fun drawEmpty(canvas: Canvas, x: Float, y: Float) {
        canvas.drawBitmap(empty, x, y, defaultPaint)
    }
}